src/NuGet.Core/NuGet.ProjectManagement/NuGetProject.cs

```diff
@@ -35,6 +37,14 @@ protected NuGetProject(Dictionary<string, object> metadata)
             get { return InternalMetadata; }
         }
 
+        public string Guid
```

Comment:
This should not exist. If you need a guid for the start/end generate one based on the operation, not the project. Consider also what might happen if we support parallel operations on a project in a future version of NuGet.